TF-360: Interaction Config screen in web settings UI

NatGreenwood-UL opened the pull request:
- [x] Test plan updated (by PR creator) 
- [x] Code Review 
- [x] Developer Testing: Key item tests 
- [x] QA Review
- [ ] Product Owner Review (Optional) 
- [ ] Update Release Notes on the JIRA ticket for harvest at release time

To access the settings UI, with a build of TouchFree Service from this branch active, access `http://localhost:9739/settings/index.html`
An autorun.brs for BrightSign testing can also be provided

NatGreenwood-UL committed:
Refactor ControlBar to shift styles out to a css file

NatGreenwood-UL committed:
Tidy up core styles css & how it's loaded to be more generally usable

NatGreenwood-UL committed:
Refactor base Page class to allow children to define their own
state/prop types

Updates CameraPage to keep its existing state.

InteractionsPage is the main focus of this story so that will come in a
later commit

NatGreenwood-UL committed:
Add the Interaction Preview videos & supporting files to get React to
load them

NatGreenwood-UL committed:
Interaction Selection UI (not connected to Service)

NatGreenwood-UL committed:
Merge branch 'develop' into feature/TF-360_Interaction_Setup_Screen_in_web_settings_UI

NatGreenwood-UL committed:
Add all of the controls, set a couple up to prove they work

NatGreenwood-UL committed:
Try using raw TS for TF Tooling to allow proper importing

NatGreenwood-UL committed:
Some small tweaks to tooling to allow the use of TF Tooling as
TypeScript

NatGreenwood-UL committed:
Update tooling used in Settings UI & add dependent package

NatGreenwood-UL committed:
Fully connect UI Logic, save to files and match sliders to Unity UI

NatGreenwood-UL committed:
Remove unused imports & fix comparisons in Settings UI & Tooling Itself

NatGreenwood-UL committed:
Change to flex-end

PeterNancollis-UH commented on TF_Settings_Web/src/Components/Pages/InteractionsPage.tsx:
```diff
@@ -0,0 +1,307 @@
+                        <Slider name="Cursor Movement"
+                            increment={0.001}
+                            rangeMin={0}
+                            rangeMax={1}
```
I think this rangeMax value is setting a deadzone in meters where it should be centimeters. With the slider at half way I have to move my hand about 0.5m to make the cursor move

NatGreenwood-UL replied:
This is matched to the range of the slider in Unity now 👌

PeterNancollis-UH commented on TF_Settings_Web/src/Components/Pages/InteractionsPage.tsx:
```diff
@@ -0,0 +1,307 @@
+
+    resetToDefaults(): void {
+        // ???????
+        // How to get an "Defaults?"
```
Is it intentional that this isn't implemented?

NatGreenwood-UL replied:
This is now implemented

PeterNancollis-UH commented on TF_Settings_Web/src/App.css:
```diff
@@ -0,0 +1,30 @@
+.App {
```
Not sure the correct place to comment on this, but the app doesn't fill the vertical height of the browser. I think it should fill with the background colour to the full height even when there's only content at the top
![IUgS2pkyJF](https://user-images.githubusercontent.com/52566990/160155179-a3e4e0f0-2daa-4f26-8946-4792d4869b49.png)

NatGreenwood-UL committed:
Add default values & implement Reset to Defaults button

Also switch to ... notation for combining old state with new (for
one-level deep additions)

NatGreenwood-UL committed:
Add missing file from last commit

NatGreenwood-UL committed:
Switch to using the state referenced passed into setState

NatGreenwood-UL committed:
Tidy some UI and improve event handling

Still trying to find root cause of double-clicks
UI still doesn't respond to Pointer events

GabrielHomewoodUL committed:
Update web settings web tooling with fixes for cursor scaling

GabrielHomewoodUL committed:
TF-360 Fix duplicate input actions being handled by webinputcontroller

NatGreenwood-UL committed:
Move connection to touchfree to root app & tweak ServiceConnection to
try and minimise connection attempts

NatGreenwood-UL committed:
Enable all controls to be used with TouchFree

NatGreenwood-UL committed:
Remove Snapping plugin & fix gitignore

NatGreenwood-UL committed:
Strip debug info from Tooling

NatGreenwood-UL committed:
Unite Tooling in Settings UI with regular tooling

PeterNancollis-UH commented on TF_Settings_Web/src/TouchFree/Configuration/ConfigurationManager.ts:
```diff
@@ -0,0 +1,101 @@
+import {
```
Is this a copy of the source for tooling to add it to the settings UI? 

I know we've talked about automating the maintaining of this but just wanted to check

NatGreenwood-UL replied:
Yes, the `src/TouchFree` directory is a clone of the Web Tooling for now (until we do the work you mentioned). I've made sure that the two are identical as of the last commit I made but this is definitely another motivator to _either_ do the automatic import, to work out what the workflow for people who want to consume tooling as TypeScript looks like, or possibly both?

PeterNancollis-UH replied:
I think both are important going forward. This is fine for the time being though

PeterNancollis-UH commented on TF_Tooling_Web/.gitignore:
```diff
@@ -8,4 +8,4 @@
 dist
 
-./Plugins/**/SnappingPlugin.js
+./Plugins/**/Snapping_Plugin.js
```
Why has the snapping plugin been added to gitignore? Is this just removing a locally built version?

NatGreenwood-UL replied:
Yes, the .js form is the built version. The change here is from an old name to a new name, it was always meant to be ignored.

NatGreenwood-UL committed:
Merge branch 'develop' into feature/TF-360_Interaction_Setup_Screen_in_web_settings_UI

NatGreenwood-UL committed:
Fix unit capitalisation in TouchPlane variable, add keys where needed

PeterNancollis-UH commented on TF_Settings_Web/src/Components/Controls/Slider.tsx:
```diff
@@ -0,0 +1,87 @@
+                        onChange={this.onChange}
+                        onPointerMove={this.onMove.bind(this)}
+                        onPointerDown={this.onDown.bind(this)}
+                        onPointerUp={this.onUpOut.bind(this)}
```
I think we need to add an `onPointerLeave` handler calling the `onUpOut` method to prevent the handle getting stuck to the cursor

NatGreenwood-UL committed:
Make app fill screen vertically & allow releasing the slider whilst not
on it

NatGreenwood-UL committed:
Limit sliders to the ranges provided & improve slider "release"

NatGreenwood-UL committed:
Ensure reset to defaults button can be pushed by TF

NatGreenwood-UL committed:
Change sliders to handle the width of the thumb control

Unfortunately isn't dynamic, will need to be changed based on any future
changes to the styling of the slider.

PeterNancollis-UH reviewed (approved):
Tested on BrightSign. All controls can be operated by mouse and TouchFree and although there are a few improvements we still want to make we can deal with those as separate work as this does provide a usable way to configure the interactions as is
